src/redisraft.c

```diff
@@ -574,6 +594,14 @@ static void handleRedisCommand(RedisRaftCtx *rr,
         return;
     }
 
+    /* update cmds array with the client's saved "asking" state */
+    cmds->asking = getAskingState(rr, ctx);
```

Comment:
I realized one thing, what is our "asking" policy about the proxy? If these commands come via proxy, "asking" might be true here already and we'll overwrite it back to "false". This might be okay I guess, in the worst case is the request will be rejected. So, either we need a comment about proxy or we should not overwrite this value if is already set? Something like:

```c
/* Comment describing proxy case */
if (!cmds->asking) {
     cmds->asking = getAskingState(rr, ctx);
}
```